pkg/operator/config.go

```diff
@@ -228,5 +235,15 @@ func ConfigFromEnvironment() *Config {
 		config.WebhooksCertDir = webhooksCertDir
 	}
 
+	if metricsPort, ok := os.LookupEnv("METRICS_PORT"); ok {
+		v, err := strconv.ParseInt(metricsPort, 10, 32)
+		if err != nil {
+			v = DefaultMetricsPort
+			klog.Errorf("Error parsing METRICS_PORT environment variable: %v", err)
+		}
+
+		config.metricsPort = int(v)
```

Comment:
If the `ParseInt` fails then metricsPort is likely to (the zero value) 0. This will then listen on port 0 -- the ephemeral port number - and will appear to work as the OS will assign an unused port number. I don't think we want this. We should return an error if the conversion fails.